### ISSUE TITLE
UseRequestHeadersForMetadataAddressBehavior not working with NetTcpBinding

### ISSUE THREAD

**Issue by hav00k** [CoreWCF/CoreWCF#727]
It seems that the request headers are ignored when generating WSDL for a `NetTcpBinding`.

I have a net.tcp service listening on `net.tcp://0.0.0.0:8090` and a help page on `http://0.0.0.0:8080/mex`, which when accessed from a browser on `http://localhost:8080/mex` correctly shows the WSDL link as `http://localhost:8080/mex?wsdl`.

But the generated WSDL ignores the host and instead contains the following:

`<wsa10:Address>net.tcp://0.0.0.0:8090/My</wsa10:Address>`

This is not the case with `BasicHttpBinding` where this functionality works as expected.

**Comment by mconnew** [comment]
Does the solution in #713 work for you?

**Comment by hav00k** [comment]
No, unfortunately setting the service base address makes no difference.

In this case the port is actually correct, it's the ip address/hostname that doesn't update according to the request headers from the help page. It stays to whatever was set in the base address (<any ip> as in the example above).
